apps/studio/data/table-rows/table-rows-query.ts

```diff
@@ -133,11 +133,14 @@ export const getTableRowsSqlQuery = ({
 
   if (!table) return ``
 
-  const arrayBasedColumns = table.columns
-    .filter(
-      (column) => (column?.enum ?? []).length > 0 && column.dataType.toLowerCase() === 'array'
-    )
-    .map((column) => `"${column.name}"::text[]`)
+  const arrayBasedColumns =
+    table.columns.length > 0
+      ? table.columns
+          .filter(
+            (column) => (column?.enum ?? []).length > 0 && column.dataType.toLowerCase() === 'array'
+          )
+          .map((column) => `"${column.name}"::text[]`)
+      : []
```

Comment:
nit: this one's an unnecessary check cause the filter will either way return an empty array (and the subsequent map will also return an empty array)